src/main/java/hudson/plugins/promoted_builds/pipeline/PipelinePromotionCondition.java

```diff
@@ -0,0 +1,52 @@
+package hudson.plugins.promoted_builds.pipeline;
+
+import hudson.DescriptorExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Describable;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.promoted_builds.PromotionBadge;
+import hudson.plugins.promoted_builds.util.JenkinsHelper;
+
+import javax.annotation.CheckForNull;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class PipelinePromotionCondition implements ExtensionPoint, Describable<PipelinePromotionCondition> {
```

Comment:
There is no need in separate extension point. The existing ones should be refactored as we agreed before